content/kotlin/concepts/loops/loops.md

```diff
@@ -0,0 +1,108 @@
+---
+Title: 'Loops'
+Description: 'In kotlin, a loop is a control structure that allows you to repeat a block of code a specified number of times or until a certain condition is met. Loops are useful for automating repetitive tasks and for processing large amounts of data efficiently.'
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'loop'
+  - 'while'
+  - 'for'
+  - 'do-while'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+In Kotlin, a loop is a control flow statement that allows you to repeat a block of code a certain number of times or until a certain condition is met. Kotlin provides several types of loops, including:
+
+- `for` loop: This loop iterates over a range of values or elements in a collection.
+- `while` loop: This loop repeatedly executes a block of code as long as a certain condition is true.
+- `do-while` loop: This loop is similar to a while loop, but it guarantees that the block of code will be executed at least once before the condition is checked.
+
+## While loop
+
+The `while` loops used to execute a block of code repeatedly as long as a certain condition is true. The syntax for a `while` loop in Kotlin is:
+
+```kotlin
+    fun main(){
+        while(condition){
+            //code to be execute while the condition is true
+        }
+    }
+```
+
+Here's an example of a `while` loop in Kotlin that prints the numbers from 1 to 10:
+
+```kotlin
+    fun main(){
+    var i = 1
+    while (i <= 10) {
+    print(" " + i)
+    i++
+    }
+}
+```
+
+The output for the above code will be:
+
+```
+1 2 3 4 5 6 7 8 9 10
+```
+
+## For loop
+
+The `for` loop is used to iterate over a range of values or an iterable collection. The syntax for a `for` loop in Kotlin is:
+
+```kotlin
+fun main(){
+    for (item in collection) {
+   // code to be executed for each item
+    }
+}
+```
+
+Here's an example of a `for` loop in Kotlin that prints the numbers from 1 to 10:
+
+```kotlin
+fun main(){
+    for(i in 1..10){
+        print(i)
+    }
+}
+```
+
+The output for the above code will be:
+
+```
+1 2 3 4 5 6 7 8 9 10
+```
+
+## Do-While loop
+
+A `do-while` loop in Kotlin is similar to a `while` loop, but the block of code is always executed at least once before the condition is checked. The syntax for a `do-while` loop is:
+
+```kotlin
+fun main(){
+    do {
+    // code to be executed at least once
+    } while (condition)
+}
+```
+
+Here's an example of a `do-while` loop in Kotlin that prompts the user to enter a number and prints the decreasing order sequence till 0:
```

Comment:
This is not what the example does. Either change the example to match the description, or alter the description.

Reply:
alright wait I will try to fix the description